src/briefcase/platforms/macOS/__init__.py

```diff
@@ -83,7 +84,8 @@ def run_app(self, app: BaseConfig, **kwargs):
         self.logger.info()
         self.logger.info(f"[{app.app_name}] Following system log output (type CTRL-C to stop log)...")
         self.logger.info("=" * 75)
-        self.subprocess.stream_output("log stream", log_popen)
+        app_pid = get_process_id_by_command(command=str(self.binary_path(app)))
```

Comment:
Looking for edge cases - if the process crashes quickly, `app_pid` will return None; but this will still try to display a log (and because of the behaviour of is_process_dead, will run forever). Should we perhaps check explicitly for app_pid == None (or raise an exception for "can't find PID"), and log that as a "can't stream because I can't find the app"?)

Reply:
Good call. I added an error log to indicate this. I also refactored `macOSRunMixIn` just a bit to ensure the log streaming process is terminated on exit; it is safe to call `cleanup` multiple times or after the process exits itself.